In [1]:
import pandas as pd

# 1) Loading the dataset 

In [2]:
raw_data = pd.read_csv('Absenteeism_data.csv')
raw_data

ID  Reason for Absence        Date  Transportation Expense  \
0    11                  26  07/07/2015                     289   
1    36                   0  14/07/2015                     118   
2     3                  23  15/07/2015                     179   
3     7                   7  16/07/2015                     279   
4    11                  23  23/07/2015                     289   
..   ..                 ...         ...                     ...   
695  17                  10  23/05/2018                     179   
696  28                   6  23/05/2018                     225   
697  18                  10  24/05/2018                     330   
698  25                  23  24/05/2018                     235   
699  15                  28  31/05/2018                     291   

     Distance to Work  Age  Daily Work Load Average  Body Mass Index  \
0                  36   33                  239.554               30   
1                  13   50                  239.554               31   
2                  51   38                  239.554               31   
3                   5   39                  239.554               24   
4                  36   33                  239.554               30   
..                ...  ...                      ...              ...   
695                22   40                  237.656               22   
696                26   28                  237.656               24   
697                16   28                  237.656               25   
698                16   32                  237.656               25   
699                31   40                  237.656               25   

     Education  Children  Pets  Absenteeism Time in Hours  
0            1         2     1                          4  
1            1         1     0                          0  
2            1         0     0                          2  
3            1         2     0                          4  
4            1         2     1                          2  
..         ...       ...   ...                        ...  
695          2         2     0                          8  
696          1         1     2                          3  
697          2         0     0                          8  
698          3         0     0                          2  
699          1         1     1                          2  

[700 rows x 12 columns]

In [3]:
 pd.options.display.max_columns = None

In [4]:
df = raw_data.copy()
df

ID  Reason for Absence        Date  Transportation Expense  \
0    11                  26  07/07/2015                     289   
1    36                   0  14/07/2015                     118   
2     3                  23  15/07/2015                     179   
3     7                   7  16/07/2015                     279   
4    11                  23  23/07/2015                     289   
..   ..                 ...         ...                     ...   
695  17                  10  23/05/2018                     179   
696  28                   6  23/05/2018                     225   
697  18                  10  24/05/2018                     330   
698  25                  23  24/05/2018                     235   
699  15                  28  31/05/2018                     291   

     Distance to Work  Age  Daily Work Load Average  Body Mass Index  \
0                  36   33                  239.554               30   
1                  13   50                  239.554               31   
2                  51   38                  239.554               31   
3                   5   39                  239.554               24   
4                  36   33                  239.554               30   
..                ...  ...                      ...              ...   
695                22   40                  237.656               22   
696                26   28                  237.656               24   
697                16   28                  237.656               25   
698                16   32                  237.656               25   
699                31   40                  237.656               25   

     Education  Children  Pets  Absenteeism Time in Hours  
0            1         2     1                          4  
1            1         1     0                          0  
2            1         0     0                          2  
3            1         2     0                          4  
4            1         2     1                          2  
..         ...       ...   ...                        ...  
695          2         2     0                          8  
696          1         1     2                          3  
697          2         0     0                          8  
698          3         0     0                          2  
699          1         1     1                          2  

[700 rows x 12 columns]

In [5]:
df.describe()

ID  Reason for Absence  Transportation Expense  \
count  700.000000          700.000000              700.000000   
mean    17.951429           19.411429              222.347143   
std     11.028144            8.356292               66.312960   
min      1.000000            0.000000              118.000000   
25%      9.000000           13.000000              179.000000   
50%     18.000000           23.000000              225.000000   
75%     28.000000           27.000000              260.000000   
max     36.000000           28.000000              388.000000   

       Distance to Work         Age  Daily Work Load Average  Body Mass Index  \
count        700.000000  700.000000               700.000000       700.000000   
mean          29.892857   36.417143               271.801774        26.737143   
std           14.804446    6.379083                40.021804         4.254701   
min            5.000000   27.000000               205.917000        19.000000   
25%           16.000000   31.000000               241.476000        24.000000   
50%           26.000000   37.000000               264.249000        25.000000   
75%           50.000000   40.000000               294.217000        31.000000   
max           52.000000   58.000000               378.884000        38.000000   

        Education    Children        Pets  Absenteeism Time in Hours  
count  700.000000  700.000000  700.000000                 700.000000  
mean     1.282857    1.021429    0.687143                   6.761429  
std      0.668090    1.112215    1.166095                  12.670082  
min      1.000000    0.000000    0.000000                   0.000000  
25%      1.000000    0.000000    0.000000                   2.000000  
50%      1.000000    1.000000    0.000000                   3.000000  
75%      1.000000    2.000000    1.000000                   8.000000  
max      4.000000    4.000000    8.000000                 120.000000

In [6]:
df.isnull().sum()

ID                           0
Reason for Absence           0
Date                         0
Transportation Expense       0
Distance to Work             0
Age                          0
Daily Work Load Average      0
Body Mass Index              0
Education                    0
Children                     0
Pets                         0
Absenteeism Time in Hours    0
dtype: int64

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   ID                         700 non-null    int64  
 1   Reason for Absence         700 non-null    int64  
 2   Date                       700 non-null    object 
 3   Transportation Expense     700 non-null    int64  
 4   Distance to Work           700 non-null    int64  
 5   Age                        700 non-null    int64  
 6   Daily Work Load Average    700 non-null    float64
 7   Body Mass Index            700 non-null    int64  
 8   Education                  700 non-null    int64  
 9   Children                   700 non-null    int64  
 10  Pets                       700 non-null    int64  
 11  Absenteeism Time in Hours  700 non-null    int64  
dtypes: float64(1), int64(10), object(1)
memory usage: 65.8+ KB


In [8]:
df = df.drop(['ID'],axis = 1)

In [9]:
df

Reason for Absence        Date  Transportation Expense  Distance to Work  \
0                    26  07/07/2015                     289                36   
1                     0  14/07/2015                     118                13   
2                    23  15/07/2015                     179                51   
3                     7  16/07/2015                     279                 5   
4                    23  23/07/2015                     289                36   
..                  ...         ...                     ...               ...   
695                  10  23/05/2018                     179                22   
696                   6  23/05/2018                     225                26   
697                  10  24/05/2018                     330                16   
698                  23  24/05/2018                     235                16   
699                  28  31/05/2018                     291                31   

     Age  Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0     33                  239.554               30          1         2     1   
1     50                  239.554               31          1         1     0   
2     38                  239.554               31          1         0     0   
3     39                  239.554               24          1         2     0   
4     33                  239.554               30          1         2     1   
..   ...                      ...              ...        ...       ...   ...   
695   40                  237.656               22          2         2     0   
696   28                  237.656               24          1         1     2   
697   28                  237.656               25          2         0     0   
698   32                  237.656               25          3         0     0   
699   40                  237.656               25          1         1     1   

     Absenteeism Time in Hours  
0                            4  
1                            0  
2                            2  
3                            4  
4                            2  
..                         ...  
695                          8  
696                          3  
697                          8  
698                          2  
699                          2  

[700 rows x 11 columns]

# 2) Reason for Absence

In [10]:
df['Reason for Absence'].min()

0

In [11]:
len(df['Reason for Absence'].unique())

28

In [12]:
sorted(df['Reason for Absence'].unique())

[0,
 1,
 2,
 3,
 4,
 5,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 13,
 14,
 15,
 16,
 17,
 18,
 19,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28]

Min, Max values are 0 and 28, indicating there are 29 numbers respectively. Whereas the length method shows that there are only 28 numbers in 'Reason for Absence'. And the missing number is 20.

As the numbers from 0-28(except 20) in 'Reason for Absence'column describes certain types of diseases or a reason for absence we will try to get the dummy variables that represents only 0 and 1, as this will make the dataset more readable.

Reasons source:
* https://clemence5.github.io/absenteeism/
* https://www.kaggle.com/datasets/tonypriyanka2913/employee-absenteeism

In [13]:
reason_columns = pd.get_dummies(df['Reason for Absence'])

In [14]:
reason_columns

0   1   2   3   4   5   6   7   8   9   10  11  12  13  14  15  16  17  \
0     0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
1     1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
2     0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
3     0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   
4     0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..   
695   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   
696   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   
697   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   
698   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
699   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   

     18  19  21  22  23  24  25  26  27  28  
0     0   0   0   0   0   0   0   1   0   0  
1     0   0   0   0   0   0   0   0   0   0  
2     0   0   0   0   1   0   0   0   0   0  
3     0   0   0   0   0   0   0   0   0   0  
4     0   0   0   0   1   0   0   0   0   0  
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  
695   0   0   0   0   0   0   0   0   0   0  
696   0   0   0   0   0   0   0   0   0   0  
697   0   0   0   0   0   0   0   0   0   0  
698   0   0   0   0   1   0   0   0   0   0  
699   0   0   0   0   0   0   0   0   0   1  

[700 rows x 28 columns]

In [15]:
reason_columns['check'] = reason_columns.sum(axis = 1)

In [16]:
reason_columns

0  1  2  3  4  5  6  7  8  9  10  11  12  13  14  15  16  17  18  19  21  \
0    0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
1    1  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
2    0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
3    0  0  0  0  0  0  0  1  0  0   0   0   0   0   0   0   0   0   0   0   0   
4    0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
..  .. .. .. .. .. .. .. .. .. ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..   
695  0  0  0  0  0  0  0  0  0  0   1   0   0   0   0   0   0   0   0   0   0   
696  0  0  0  0  0  0  1  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
697  0  0  0  0  0  0  0  0  0  0   1   0   0   0   0   0   0   0   0   0   0   
698  0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
699  0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   

     22  23  24  25  26  27  28  check  
0     0   0   0   0   1   0   0      1  
1     0   0   0   0   0   0   0      1  
2     0   1   0   0   0   0   0      1  
3     0   0   0   0   0   0   0      1  
4     0   1   0   0   0   0   0      1  
..   ..  ..  ..  ..  ..  ..  ..    ...  
695   0   0   0   0   0   0   0      1  
696   0   0   0   0   0   0   0      1  
697   0   0   0   0   0   0   0      1  
698   0   1   0   0   0   0   0      1  
699   0   0   0   0   0   0   1      1  

[700 rows x 29 columns]

In [17]:
reason_columns['check'].sum(axis = 0)

700

In [18]:
reason_columns['check'].unique()

array([1], dtype=int64)

In [19]:
reason_columns = reason_columns.drop(['check'], axis = 1)
reason_columns

0  1  2  3  4  5  6  7  8  9  10  11  12  13  14  15  16  17  18  19  21  \
0    0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
1    1  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
2    0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
3    0  0  0  0  0  0  0  1  0  0   0   0   0   0   0   0   0   0   0   0   0   
4    0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
..  .. .. .. .. .. .. .. .. .. ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..   
695  0  0  0  0  0  0  0  0  0  0   1   0   0   0   0   0   0   0   0   0   0   
696  0  0  0  0  0  0  1  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
697  0  0  0  0  0  0  0  0  0  0   1   0   0   0   0   0   0   0   0   0   0   
698  0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   
699  0  0  0  0  0  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   

     22  23  24  25  26  27  28  
0     0   0   0   0   1   0   0  
1     0   0   0   0   0   0   0  
2     0   1   0   0   0   0   0  
3     0   0   0   0   0   0   0  
4     0   1   0   0   0   0   0  
..   ..  ..  ..  ..  ..  ..  ..  
695   0   0   0   0   0   0   0  
696   0   0   0   0   0   0   0  
697   0   0   0   0   0   0   0  
698   0   1   0   0   0   0   0  
699   0   0   0   0   0   0   1  

[700 rows x 28 columns]

The index 0 column is not required, as from the 'Reasons of Absence' list there is no number significant to 0.

In [20]:
reason_columns = pd.get_dummies(df['Reason for Absence'], drop_first = True)

In [21]:
reason_columns

1   2   3   4   5   6   7   8   9   10  11  12  13  14  15  16  17  18  \
0     0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
1     0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
2     0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
3     0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   
4     0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..   
695   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   
696   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   
697   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   
698   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
699   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   

     19  21  22  23  24  25  26  27  28  
0     0   0   0   0   0   0   1   0   0  
1     0   0   0   0   0   0   0   0   0  
2     0   0   0   1   0   0   0   0   0  
3     0   0   0   0   0   0   0   0   0  
4     0   0   0   1   0   0   0   0   0  
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  
695   0   0   0   0   0   0   0   0   0  
696   0   0   0   0   0   0   0   0   0  
697   0   0   0   0   0   0   0   0   0  
698   0   0   0   1   0   0   0   0   0  
699   0   0   0   0   0   0   0   0   1  

[700 rows x 27 columns]

### 2a) Grouping the columns 1-28 into a class

In [22]:
df.columns.values

array(['Reason for Absence', 'Date', 'Transportation Expense',
       'Distance to Work', 'Age', 'Daily Work Load Average',
       'Body Mass Index', 'Education', 'Children', 'Pets',
       'Absenteeism Time in Hours'], dtype=object)

In [23]:
reason_columns.columns.values

array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,
       18, 19, 21, 22, 23, 24, 25, 26, 27, 28], dtype=int64)

Drop the 'Reason for Absence' columns from 'df' table as the "reason_column" says conveys the same meaning. If not deleted, we might cause multicolinearity.

In [24]:
df = df.drop(['Reason for Absence'], axis = 1)


In [25]:
df

Date  Transportation Expense  Distance to Work  Age  \
0    07/07/2015                     289                36   33   
1    14/07/2015                     118                13   50   
2    15/07/2015                     179                51   38   
3    16/07/2015                     279                 5   39   
4    23/07/2015                     289                36   33   
..          ...                     ...               ...  ...   
695  23/05/2018                     179                22   40   
696  23/05/2018                     225                26   28   
697  24/05/2018                     330                16   28   
698  24/05/2018                     235                16   32   
699  31/05/2018                     291                31   40   

     Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0                    239.554               30          1         2     1   
1                    239.554               31          1         1     0   
2                    239.554               31          1         0     0   
3                    239.554               24          1         2     0   
4                    239.554               30          1         2     1   
..                       ...              ...        ...       ...   ...   
695                  237.656               22          2         2     0   
696                  237.656               24          1         1     2   
697                  237.656               25          2         0     0   
698                  237.656               25          3         0     0   
699                  237.656               25          1         1     1   

     Absenteeism Time in Hours  
0                            4  
1                            0  
2                            2  
3                            4  
4                            2  
..                         ...  
695                          8  
696                          3  
697                          8  
698                          2  
699                          2  

[700 rows x 10 columns]

In [26]:
reason_type_1 = reason_columns.loc[:, 1:14].max(axis=1)
reason_type_2 = reason_columns.loc[:, 15:17].max(axis=1)
reason_type_3 = reason_columns.loc[:, 18:21].max(axis=1)
reason_type_4 = reason_columns.loc[:, 22:].max(axis=1)

In [27]:
df = pd.concat([df, reason_type_1, reason_type_2, reason_type_3, reason_type_4], axis =1)
df

Date  Transportation Expense  Distance to Work  Age  \
0    07/07/2015                     289                36   33   
1    14/07/2015                     118                13   50   
2    15/07/2015                     179                51   38   
3    16/07/2015                     279                 5   39   
4    23/07/2015                     289                36   33   
..          ...                     ...               ...  ...   
695  23/05/2018                     179                22   40   
696  23/05/2018                     225                26   28   
697  24/05/2018                     330                16   28   
698  24/05/2018                     235                16   32   
699  31/05/2018                     291                31   40   

     Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0                    239.554               30          1         2     1   
1                    239.554               31          1         1     0   
2                    239.554               31          1         0     0   
3                    239.554               24          1         2     0   
4                    239.554               30          1         2     1   
..                       ...              ...        ...       ...   ...   
695                  237.656               22          2         2     0   
696                  237.656               24          1         1     2   
697                  237.656               25          2         0     0   
698                  237.656               25          3         0     0   
699                  237.656               25          1         1     1   

     Absenteeism Time in Hours  0  1  2  3  
0                            4  0  0  0  1  
1                            0  0  0  0  0  
2                            2  0  0  0  1  
3                            4  1  0  0  0  
4                            2  0  0  0  1  
..                         ... .. .. .. ..  
695                          8  1  0  0  0  
696                          3  1  0  0  0  
697                          8  1  0  0  0  
698                          2  0  0  0  1  
699                          2  0  0  0  1  

[700 rows x 14 columns]

### 2b) Rearranging the column names

In [28]:
df.columns.values

array(['Date', 'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours', 0, 1, 2, 3],
      dtype=object)

In [29]:
#Changing the columns names
column_names = ['Date', 'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours', 'Reason_1', 'Reason_2', 'Reason_3', 'Reason_4']
df.columns = column_names

In [30]:
df.head()

Date  Transportation Expense  Distance to Work  Age  \
0  07/07/2015                     289                36   33   
1  14/07/2015                     118                13   50   
2  15/07/2015                     179                51   38   
3  16/07/2015                     279                 5   39   
4  23/07/2015                     289                36   33   

   Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0                  239.554               30          1         2     1   
1                  239.554               31          1         1     0   
2                  239.554               31          1         0     0   
3                  239.554               24          1         2     0   
4                  239.554               30          1         2     1   

   Absenteeism Time in Hours  Reason_1  Reason_2  Reason_3  Reason_4  
0                          4         0         0         0         1  
1                          0         0         0         0         0  
2                          2         0         0         0         1  
3                          4         1         0         0         0  
4                          2         0         0         0         1

In [31]:
column_names_reordered = ['Reason_1', 'Reason_2', 'Reason_3', 'Reason_4','Date', 'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours']
df = df[column_names_reordered]

In [32]:
df.head()

Reason_1  Reason_2  Reason_3  Reason_4        Date  Transportation Expense  \
0         0         0         0         1  07/07/2015                     289   
1         0         0         0         0  14/07/2015                     118   
2         0         0         0         1  15/07/2015                     179   
3         1         0         0         0  16/07/2015                     279   
4         0         0         0         1  23/07/2015                     289   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          1   
1                13   50                  239.554               31          1   
2                51   38                  239.554               31          1   
3                 5   39                  239.554               24          1   
4                36   33                  239.554               30          1   

   Children  Pets  Absenteeism Time in Hours  
0         2     1                          4  
1         1     0                          0  
2         0     0                          2  
3         2     0                          4  
4         2     1                          2

### checkpoint 

In [33]:
#modified and classified the "Reasons of Absence" column
df_reasons_modified = df.copy()
df_reasons_modified

Reason_1  Reason_2  Reason_3  Reason_4        Date  \
0           0         0         0         1  07/07/2015   
1           0         0         0         0  14/07/2015   
2           0         0         0         1  15/07/2015   
3           1         0         0         0  16/07/2015   
4           0         0         0         1  23/07/2015   
..        ...       ...       ...       ...         ...   
695         1         0         0         0  23/05/2018   
696         1         0         0         0  23/05/2018   
697         1         0         0         0  24/05/2018   
698         0         0         0         1  24/05/2018   
699         0         0         0         1  31/05/2018   

     Transportation Expense  Distance to Work  Age  Daily Work Load Average  \
0                       289                36   33                  239.554   
1                       118                13   50                  239.554   
2                       179                51   38                  239.554   
3                       279                 5   39                  239.554   
4                       289                36   33                  239.554   
..                      ...               ...  ...                      ...   
695                     179                22   40                  237.656   
696                     225                26   28                  237.656   
697                     330                16   28                  237.656   
698                     235                16   32                  237.656   
699                     291                31   40                  237.656   

     Body Mass Index  Education  Children  Pets  Absenteeism Time in Hours  
0                 30          1         2     1                          4  
1                 31          1         1     0                          0  
2                 31          1         0     0                          2  
3                 24          1         2     0                          4  
4                 30          1         2     1                          2  
..               ...        ...       ...   ...                        ...  
695               22          2         2     0                          8  
696               24          1         1     2                          3  
697               25          2         0     0                          8  
698               25          3         0     0                          2  
699               25          1         1     1                          2  

[700 rows x 14 columns]

# 3) Date

In [34]:
#date is a string type
type(df_reasons_modified['Date'][0])

str

In [35]:
df_reasons_modified['Date'] = pd.to_datetime(df_reasons_modified['Date'], format = '%d/%m/%Y')

In [36]:
df_reasons_modified['Date']

0     2015-07-07
1     2015-07-14
2     2015-07-15
3     2015-07-16
4     2015-07-23
         ...    
695   2018-05-23
696   2018-05-23
697   2018-05-24
698   2018-05-24
699   2018-05-31
Name: Date, Length: 700, dtype: datetime64[ns]

In [37]:
type(df_reasons_modified['Date'][0])
#Now the date column is converted to timestamp

pandas._libs.tslibs.timestamps.Timestamp

In [38]:
df_reasons_modified.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 14 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   Reason_1                   700 non-null    uint8         
 1   Reason_2                   700 non-null    uint8         
 2   Reason_3                   700 non-null    uint8         
 3   Reason_4                   700 non-null    uint8         
 4   Date                       700 non-null    datetime64[ns]
 5   Transportation Expense     700 non-null    int64         
 6   Distance to Work           700 non-null    int64         
 7   Age                        700 non-null    int64         
 8   Daily Work Load Average    700 non-null    float64       
 9   Body Mass Index            700 non-null    int64         
 10  Education                  700 non-null    int64         
 11  Children                   700 non-null    int64         
 12  Pets    

### 3a) Extracting the Month Values Only from df_reasons_modified

In [39]:
df_reasons_modified['Date'][0].month

7

In [40]:
#looping over df_reasons_modified, date columns to extraxct only the month values from it.
list_months = []
for i in range(df_reasons_modified.shape[0]):
    list_months.append(df_reasons_modified['Date'][i].month)
list_months

[7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 7,
 

In [41]:
len(list_months)

700

In [42]:
#creating a new column named "Month Value" and appending it to df_reasons_modified
df_reasons_modified['Month Value'] = list_months
df_reasons_modified.head()

Reason_1  Reason_2  Reason_3  Reason_4       Date  Transportation Expense  \
0         0         0         0         1 2015-07-07                     289   
1         0         0         0         0 2015-07-14                     118   
2         0         0         0         1 2015-07-15                     179   
3         1         0         0         0 2015-07-16                     279   
4         0         0         0         1 2015-07-23                     289   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          1   
1                13   50                  239.554               31          1   
2                51   38                  239.554               31          1   
3                 5   39                  239.554               24          1   
4                36   33                  239.554               30          1   

   Children  Pets  Absenteeism Time in Hours  Month Value  
0         2     1                          4            7  
1         1     0                          0            7  
2         0     0                          2            7  
3         2     0                          4            7  
4         2     1                          2            7

Month Values always ranges from 1-12.

### 3b)  Extracting the Day Of the Week Values Only from df_reasons_modified

In [43]:
df_reasons_modified['Date'][699].weekday()

3

In [44]:
#looping over df_reasons_modified, date columns to extraxct only the day of the week values from it.
list_days_of_week = []
for i in range(df_reasons_modified.shape[0]):
    list_days_of_week.append(df_reasons_modified['Date'][i].weekday())
list_days_of_week

[1,
 1,
 2,
 3,
 3,
 4,
 4,
 4,
 0,
 0,
 0,
 1,
 2,
 2,
 2,
 4,
 4,
 0,
 3,
 2,
 2,
 0,
 0,
 4,
 0,
 0,
 1,
 2,
 2,
 4,
 0,
 3,
 3,
 0,
 0,
 0,
 1,
 3,
 4,
 4,
 1,
 0,
 1,
 1,
 2,
 6,
 0,
 3,
 4,
 0,
 0,
 0,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 4,
 4,
 4,
 1,
 1,
 2,
 3,
 4,
 4,
 1,
 2,
 2,
 3,
 2,
 2,
 3,
 1,
 1,
 2,
 4,
 4,
 3,
 2,
 3,
 3,
 3,
 0,
 0,
 0,
 2,
 4,
 2,
 2,
 4,
 4,
 0,
 1,
 2,
 3,
 1,
 1,
 2,
 2,
 3,
 4,
 1,
 2,
 3,
 4,
 1,
 2,
 4,
 4,
 4,
 2,
 0,
 1,
 1,
 2,
 3,
 3,
 4,
 0,
 1,
 1,
 2,
 3,
 4,
 0,
 1,
 1,
 1,
 2,
 3,
 3,
 0,
 1,
 3,
 4,
 2,
 2,
 3,
 3,
 4,
 4,
 0,
 1,
 0,
 1,
 1,
 1,
 3,
 0,
 1,
 2,
 3,
 1,
 1,
 1,
 1,
 2,
 2,
 3,
 4,
 4,
 0,
 0,
 0,
 1,
 3,
 3,
 4,
 4,
 0,
 1,
 2,
 0,
 0,
 1,
 2,
 2,
 3,
 4,
 4,
 0,
 2,
 2,
 1,
 2,
 3,
 4,
 4,
 2,
 3,
 4,
 4,
 4,
 4,
 4,
 1,
 2,
 2,
 3,
 4,
 0,
 2,
 3,
 0,
 1,
 1,
 2,
 0,
 1,
 2,
 2,
 2,
 2,
 2,
 1,
 2,
 4,
 0,
 0,
 1,
 3,
 4,
 0,
 0,
 0,
 2,
 4,
 1,
 2,
 3,
 4,
 0,
 3,
 1,
 2,
 4,
 4,
 1,
 3,
 4,
 0,
 1,
 0,
 1,
 2,
 3,
 0,


In [45]:
len(list_days_of_week)

700

In [46]:
#creating a new column named "Day Of the Week" and appending it to df_reasons_modified
df_reasons_modified['Day Of the Week'] = list_days_of_week
df_reasons_modified.head()

Reason_1  Reason_2  Reason_3  Reason_4       Date  Transportation Expense  \
0         0         0         0         1 2015-07-07                     289   
1         0         0         0         0 2015-07-14                     118   
2         0         0         0         1 2015-07-15                     179   
3         1         0         0         0 2015-07-16                     279   
4         0         0         0         1 2015-07-23                     289   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          1   
1                13   50                  239.554               31          1   
2                51   38                  239.554               31          1   
3                 5   39                  239.554               24          1   
4                36   33                  239.554               30          1   

   Children  Pets  Absenteeism Time in Hours  Month Value  Day Of the Week  
0         2     1                          4            7                1  
1         1     0                          0            7                1  
2         0     0                          2            7                2  
3         2     0                          4            7                3  
4         2     1                          2            7                3

### 3c) Dropping "Date" Column and rearranging the column names 

In [47]:
df_reasons_modified.columns.values

array(['Reason_1', 'Reason_2', 'Reason_3', 'Reason_4', 'Date',
       'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours', 'Month Value',
       'Day Of the Week'], dtype=object)

In [48]:
df_reasons_modified.drop(['Date'],axis = 1)

Reason_1  Reason_2  Reason_3  Reason_4  Transportation Expense  \
0           0         0         0         1                     289   
1           0         0         0         0                     118   
2           0         0         0         1                     179   
3           1         0         0         0                     279   
4           0         0         0         1                     289   
..        ...       ...       ...       ...                     ...   
695         1         0         0         0                     179   
696         1         0         0         0                     225   
697         1         0         0         0                     330   
698         0         0         0         1                     235   
699         0         0         0         1                     291   

     Distance to Work  Age  Daily Work Load Average  Body Mass Index  \
0                  36   33                  239.554               30   
1                  13   50                  239.554               31   
2                  51   38                  239.554               31   
3                   5   39                  239.554               24   
4                  36   33                  239.554               30   
..                ...  ...                      ...              ...   
695                22   40                  237.656               22   
696                26   28                  237.656               24   
697                16   28                  237.656               25   
698                16   32                  237.656               25   
699                31   40                  237.656               25   

     Education  Children  Pets  Absenteeism Time in Hours  Month Value  \
0            1         2     1                          4            7   
1            1         1     0                          0            7   
2            1         0     0                          2            7   
3            1         2     0                          4            7   
4            1         2     1                          2            7   
..         ...       ...   ...                        ...          ...   
695          2         2     0                          8            5   
696          1         1     2                          3            5   
697          2         0     0                          8            5   
698          3         0     0                          2            5   
699          1         1     1                          2            5   

     Day Of the Week  
0                  1  
1                  1  
2                  2  
3                  3  
4                  3  
..               ...  
695                2  
696                2  
697                3  
698                3  
699                3  

[700 rows x 15 columns]

In [49]:
#Rearranging the column names
columns_ordered = ['Reason_1', 'Reason_2', 'Reason_3', 'Reason_4', 'Month Value','Day Of the Week',
       'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours']
df_reasons_modified = df_reasons_modified[columns_ordered]

In [50]:
df_reasons_modified.head()

Reason_1  Reason_2  Reason_3  Reason_4  Month Value  Day Of the Week  \
0         0         0         0         1            7                1   
1         0         0         0         0            7                1   
2         0         0         0         1            7                2   
3         1         0         0         0            7                3   
4         0         0         0         1            7                3   

   Transportation Expense  Distance to Work  Age  Daily Work Load Average  \
0                     289                36   33                  239.554   
1                     118                13   50                  239.554   
2                     179                51   38                  239.554   
3                     279                 5   39                  239.554   
4                     289                36   33                  239.554   

   Body Mass Index  Education  Children  Pets  Absenteeism Time in Hours  
0               30          1         2     1                          4  
1               31          1         1     0                          0  
2               31          1         0     0                          2  
3               24          1         2     0                          4  
4               30          1         2     1                          2

### checkpoint

In [51]:
df_reason_date_modified = df_reasons_modified.copy()
df_reason_date_modified.head()

Reason_1  Reason_2  Reason_3  Reason_4  Month Value  Day Of the Week  \
0         0         0         0         1            7                1   
1         0         0         0         0            7                1   
2         0         0         0         1            7                2   
3         1         0         0         0            7                3   
4         0         0         0         1            7                3   

   Transportation Expense  Distance to Work  Age  Daily Work Load Average  \
0                     289                36   33                  239.554   
1                     118                13   50                  239.554   
2                     179                51   38                  239.554   
3                     279                 5   39                  239.554   
4                     289                36   33                  239.554   

   Body Mass Index  Education  Children  Pets  Absenteeism Time in Hours  
0               30          1         2     1                          4  
1               31          1         1     0                          0  
2               31          1         0     0                          2  
3               24          1         2     0                          4  
4               30          1         2     1                          2

# 4) Education

In [52]:
df_reason_date_modified['Education'].unique()

array([1, 3, 2, 4], dtype=int64)

In "Education" the numbers 1, 2, 3, 4 represents as follows
* 1 : High School
* 2 : Graduate
* 3 : Postgraduate
* 4 : A master or doctorate

In [53]:
df_reason_date_modified['Education'].value_counts()

1    583
3     73
2     40
4      4
Name: Education, dtype: int64

Value 1 record showing that there are 583 high school greaduates alone, and over 110 candidates with  Graduate, Postgraduate, A master or doctorate degree combined. So it is an optimal way to combine these three records together

In [54]:
df_reason_date_modified['Education'] = df_reason_date_modified['Education'].map({1:0 , 2:1, 3:1, 4:1})

In [55]:
df_reason_date_modified['Education'].unique()

array([0, 1], dtype=int64)

In [56]:
df_reason_date_modified['Education'].value_counts()

0    583
1    117
Name: Education, dtype: int64

In [57]:
df_reason_date_modified.head(20)

Reason_1  Reason_2  Reason_3  Reason_4  Month Value  Day Of the Week  \
0          0         0         0         1            7                1   
1          0         0         0         0            7                1   
2          0         0         0         1            7                2   
3          1         0         0         0            7                3   
4          0         0         0         1            7                3   
5          0         0         0         1            7                4   
6          0         0         0         1            7                4   
7          0         0         0         1            7                4   
8          0         0         1         0            7                0   
9          0         0         0         1            7                0   
10         1         0         0         0            7                0   
11         1         0         0         0            7                1   
12         1         0         0         0            7                2   
13         1         0         0         0            7                2   
14         0         0         0         1            7                2   
15         1         0         0         0            7                4   
16         0         0         0         1            7                4   
17         0         0         1         0            7                0   
18         1         0         0         0            7                3   
19         0         0         0         1            8                2   

    Transportation Expense  Distance to Work  Age  Daily Work Load Average  \
0                      289                36   33                  239.554   
1                      118                13   50                  239.554   
2                      179                51   38                  239.554   
3                      279                 5   39                  239.554   
4                      289                36   33                  239.554   
5                      179                51   38                  239.554   
6                      361                52   28                  239.554   
7                      260                50   36                  239.554   
8                      155                12   34                  239.554   
9                      235                11   37                  239.554   
10                     260                50   36                  239.554   
11                     260                50   36                  239.554   
12                     260                50   36                  239.554   
13                     179                51   38                  239.554   
14                     179                51   38                  239.554   
15                     246                25   41                  239.554   
16                     179                51   38                  239.554   
17                     179                51   38                  239.554   
18                     189                29   33                  239.554   
19                     248                25   47                  205.917   

    Body Mass Index  Education  Children  Pets  Absenteeism Time in Hours  
0                30          0         2     1                          4  
1                31          0         1     0                          0  
2                31          0         0     0                          2  
3                24          0         2     0                          4  
4                30          0         2     1                          2  
5                31          0         0     0                          2  
6                27          0         1     4                          8  
7                23          0         4     0                          4  
8                25          0         2     0                         40  
9       

### checkpoint

In [58]:
df_preprocessed = df_reason_date_modified.copy()
df_preprocessed

Reason_1  Reason_2  Reason_3  Reason_4  Month Value  Day Of the Week  \
0           0         0         0         1            7                1   
1           0         0         0         0            7                1   
2           0         0         0         1            7                2   
3           1         0         0         0            7                3   
4           0         0         0         1            7                3   
..        ...       ...       ...       ...          ...              ...   
695         1         0         0         0            5                2   
696         1         0         0         0            5                2   
697         1         0         0         0            5                3   
698         0         0         0         1            5                3   
699         0         0         0         1            5                3   

     Transportation Expense  Distance to Work  Age  Daily Work Load Average  \
0                       289                36   33                  239.554   
1                       118                13   50                  239.554   
2                       179                51   38                  239.554   
3                       279                 5   39                  239.554   
4                       289                36   33                  239.554   
..                      ...               ...  ...                      ...   
695                     179                22   40                  237.656   
696                     225                26   28                  237.656   
697                     330                16   28                  237.656   
698                     235                16   32                  237.656   
699                     291                31   40                  237.656   

     Body Mass Index  Education  Children  Pets  Absenteeism Time in Hours  
0                 30          0         2     1                          4  
1                 31          0         1     0                          0  
2                 31          0         0     0                          2  
3                 24          0         2     0                          4  
4                 30          0         2     1                          2  
..               ...        ...       ...   ...                        ...  
695               22          1         2     0                          8  
696               24          0         1     2                          3  
697               25          1         0     0                          8  
698               25          1         0     0                          2  
699               25          0         1     1                          2  

[700 rows x 15 columns]

In [59]:
#storing the final dataset into .csv format
df_preprocessed.to_csv('Absenteeism_preprocessed.csv', index=False)